consult.el

```diff
@@ -1044,6 +1044,55 @@ The arguments and expected return value are as specified for
         (funcall exit completion exit-status))
       t)))
 
+(defun consult--extract-commands (mode)
+  "Extract commands from MODE."
+  (let ((library-path (symbol-file mode))
+        (mode-rx (regexp-quote
+                  (substring (if (eq major-mode 'c-mode)
+                                 "cc-mode"
+                               (symbol-name major-mode))
+                             0 -5))))
+    (mapcan
+     (lambda (feature)
+       (let ((path (car feature)))
+         (when (or (string= path library-path)
+                   (string-match-p mode-rx (file-name-nondirectory path)))
```

Comment:
But the file name check is very ad-hoc. I also didn't have a better idea yet :(

Since smex and amx both had this maybe it is a good enough heuristic which has been tested a lot?

Reply:
Yeah, it is probably is battle-tested. The `(string= path library-path)` makes sense. I wonder what typical cases the `string-match-p` catches.

Reply:
I am sure you need this if larger modes are distributed over multiple files?

Reply:
Yes, that's probably it.